lib/cn/other_iterator.c

```diff
@@ -0,0 +1,82 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <hse_util/element_source.h>
+#include <hse_util/event_counter.h>
+#include <hse_util/hse_err.h>
+
+#include "kvset.h"
+#include "kvset_internal.h"
+#include "omf.h"
+
+struct other_iterator {
+    struct kvset *oi_ks;
+    bool oi_eof;
+    uint32_t oi_offset;
+    struct element_source oi_es;
+};
+
+static bool
+next(struct element_source *source, void **data)
+{
+    struct other_iterator *iter = container_of(source, struct other_iterator, oi_es);
+
+    if (iter->oi_eof)
+        return false;
+
+    *data = &iter->oi_ks->ks_kblks[--iter->oi_offset];
+
+    if (iter->oi_offset == 0)
+        iter->oi_eof = true;
+
+    return iter->oi_eof;
+}
+
+merr_t
+other_iterator_create(
+    struct kvset *const ks,
+    struct other_iterator **const iter)
+{
+    struct other_iterator *tmp;
+
+    *iter = NULL;
+
+    tmp = calloc(1, sizeof(*tmp));
+    if (ev(!tmp))
+        return merr(ENOMEM);
+
+    /* Other members initialized to 0 in calloc() */
```

Comment:
we get that.  don't need to say it. :)

Reply:
But what if someone comes along and changes it to malloc()? :O. I will remove in the next iteration.